llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -8456,6 +8458,21 @@ LegalizerHelper::lowerAbsToCNeg(MachineInstr &MI) {
   return Legalized;
 }
 
+LegalizerHelper::LegalizeResult LegalizerHelper::lowerFAbs(MachineInstr &MI) {
+  Register SrcReg = MI.getOperand(1).getReg();
+  Register DstReg = MI.getOperand(0).getReg();
+
+  LLT Ty = MRI.getType(DstReg);
+
+  // Reset sign bit
+  MIRBuilder.buildAnd(DstReg, SrcReg,
+                      MIRBuilder.buildConstant(
+                          Ty, APInt::getSignedMaxValue(Ty.getSizeInBits())));
```

Comment:
Not 100% sure this works for ppc_fp128 but we don't support that yet anyway

Reply:
`Ty.getScalarSizeIniBts()` to support vectors? And vector support is not testet.

Reply:
@tschuett How do I test vectors? Is there a way to unit-test the helper or do I need to find a target which does use the helper for vectors (AArch64 does not).

But if we remove the `scalarizeIf` in AArch64 then `AArch64` will use the helper and hence this can be tested

Reply:
Correct.